homeassistant/components/generic/config_flow.py

```diff
@@ -0,0 +1,318 @@
+
+from homeassistant.components.stream.const import SOURCE_TIMEOUT
+from homeassistant.config_entries import ConfigEntry, ConfigFlow, OptionsFlow
+from homeassistant.const import (
+    CONF_AUTHENTICATION,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_USERNAME,
+    CONF_VERIFY_SSL,
+    HTTP_BASIC_AUTHENTICATION,
+    HTTP_DIGEST_AUTHENTICATION,
+)
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import TemplateError
+from homeassistant.helpers import config_validation as cv, template as template_helper
+from homeassistant.helpers.httpx_client import get_async_client
+
+from .camera import generate_auth
+from .const import (
+    CONF_CONTENT_TYPE,
+    CONF_FRAMERATE,
+    CONF_LIMIT_REFETCH_TO_URL_CHANGE,
+    CONF_RTSP_TRANSPORT,
+    CONF_STILL_IMAGE_URL,
+    CONF_STREAM_SOURCE,
+    DEFAULT_NAME,
+    DOMAIN,
+    FFMPEG_OPTION_MAP,
+    GET_IMAGE_TIMEOUT,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_DATA = {
+    CONF_NAME: DEFAULT_NAME,
+    CONF_AUTHENTICATION: HTTP_BASIC_AUTHENTICATION,
+    CONF_LIMIT_REFETCH_TO_URL_CHANGE: False,
+    CONF_FRAMERATE: 2,
+    CONF_VERIFY_SSL: True,
+}
+
+SUPPORTED_IMAGE_TYPES = ["png", "jpeg", "svg+xml"]
+
+
+def build_schema(user_input):
+    """Create schema for camera config setup."""
+    spec = {
+        vol.Optional(
+            CONF_STILL_IMAGE_URL,
+            description={"suggested_value": user_input.get(CONF_STILL_IMAGE_URL, "")},
+        ): str,
+        vol.Optional(
+            CONF_STREAM_SOURCE,
+            description={"suggested_value": user_input.get(CONF_STREAM_SOURCE, "")},
+        ): str,
+        vol.Optional(
+            CONF_RTSP_TRANSPORT, default=user_input.get(CONF_RTSP_TRANSPORT)
+        ): vol.In([None, "tcp", "udp", "udp_multicast", "http"]),
+        vol.Optional(
+            CONF_AUTHENTICATION, default=user_input.get(CONF_AUTHENTICATION)
+        ): vol.In([HTTP_BASIC_AUTHENTICATION, HTTP_DIGEST_AUTHENTICATION]),
+        vol.Optional(
+            CONF_USERNAME,
+            description={"suggested_value": user_input.get(CONF_USERNAME, "")},
+        ): str,
+        vol.Optional(
+            CONF_PASSWORD,
+            description={"suggested_value": user_input.get(CONF_PASSWORD, "")},
+        ): str,
+        vol.Required(
+            CONF_LIMIT_REFETCH_TO_URL_CHANGE,
+            default=user_input.get(CONF_LIMIT_REFETCH_TO_URL_CHANGE, False),
+        ): bool,
+        vol.Required(
+            CONF_FRAMERATE,
+            description={"suggested_value": user_input.get(CONF_FRAMERATE, 2)},
+        ): int,
+        vol.Required(
+            CONF_VERIFY_SSL, default=user_input.get(CONF_VERIFY_SSL, True)
+        ): bool,
+    }
+    return vol.Schema(spec)
+
+
+def get_image_type(image):
+    """Get the format of downloaded bytes that could be an image."""
+    fmt = imghdr.what(None, h=image)
+    if fmt is None:
+        # if imghdr can't figure it out, could be svg.
+        try:
```

Comment:
You could use with contextlib.suppress here

Reply:
Done